### PR TITLE
fix: sorting large data

### PR DESCRIPTION
**Pull request by howar31**

### CODE REVIEW

File: storage/news_v2.go [hunk 1]
```diff
@@ -117,7 +118,7 @@ func (m *mongoStorage) getMetaOfPosts(ctx context.Context, stages []bson.D) <-ch
 	result := make(chan fetchResult)
 	go func(ctx context.Context, stages []bson.D) {
 		defer close(result)
-		cursor, err := m.Database(globals.Conf.DB.Mongo.DBname).Collection(news.ColPosts).Aggregate(ctx, stages)
+		cursor, err := m.Database(globals.Conf.DB.Mongo.DBname).Collection(news.ColPosts).Aggregate(ctx, stages, options.Aggregate().SetAllowDiskUse(true))
```
Comment (Aylie-Chou) [inline storage/news_v2.go:121]: Question: do you have any idea why PR #509 change makes memory usage exceeding limit ?

Comment (howar31) [reply storage/news_v2.go:121]: Current guess is that we used more $group (aggregation) to get category_set data.  need further inspection for performance tweaks.

Comment (Aylie-Chou) [reply storage/news_v2.go:121]: Got it.
I would report to PM & add the research ticket for this issue.

**Review by Aylie-Chou — APPROVED** [review]
LGTM

**Pull request merged**